File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/GenerationsCoreStats.java
package generations.gg.generations.core.generationscore.common.world.level.block;

import dev.architectury.registry.registries.DeferredRegister;
import generations.gg.generations.core.generationscore.common.GenerationsCore;
import net.minecraft.core.registries.Registries;
import net.minecraft.resources.ResourceLocation;

public class GenerationsCoreStats {
    public static final DeferredRegister<ResourceLocation> STATS = DeferredRegister.create(GenerationsCore.MOD_ID, Registries.CUSTOM_STAT);

    public final static ResourceLocation HIDDEN_LOOT_FOUND = register("hidden_loot_found");
    public final static ResourceLocation NORMAL_LOOT_FOUND = register("normal_loot_found");

    private static ResourceLocation register(String name) {
        var location = GenerationsCore.id(name);
        STATS.register(name, () -> location);
        return location;
    }

    public static void init() {
        STATS.register();
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/GenerationsCoreStats.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import generations.gg.generations.core.generationscore.common.generationsResource
import generations.gg.generations.core.generationscore.common.util.PlatformRegistry
import net.minecraft.core.Holder
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.ResourceLocation

object GenerationsCoreStats: PlatformRegistry<ResourceLocation>() {
    override val registry: Registry<ResourceLocation> = BuiltInRegistries.CUSTOM_STAT
    override val resourceKey: ResourceKey<Registry<ResourceLocation>> = Registries.CUSTOM_STAT

    val HIDDEN_LOOT_FOUND = register("hidden_loot_found")
    val NORMAL_LOOT_FOUND = register("normal_loot_found")

    private fun register(name: String): Holder<ResourceLocation> = name.let { create(it, { it.generationsResource() }) }
}
